### ISSUE TITLE
Recipes depend on core-js@2

### ISSUE THREAD

**Issue by freiksenet** [gatsbyjs/gatsby#25184]
We are upgrading to core-js@3, but recipes still depend on libraries that depend on core-js@2, triggering a warning.

```
warning gatsby > gatsby-cli > gatsby-recipes > graphql-tools > @graphql-tools/relay-operation-optimizer > relay-compiler > fbjs > core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
```

Recipes should be updated to not use deprecated libraries.

**Comment by polarathene** [comment]
It's not so much the fault of the recipes. It's the long dependency chain where the issue is. The main culprit is `fbjs` which has merged a PR to update to a newer core-js, it just didn't get a new update pushed out(there hasn't been one for around 2 years).

Either that gets updated and any parents if the semver needs to accommodate, or you'll need to identify which package needs an alternative one. Replacing graphql-tools is presumably not an option, nor is forking it and dependents all the way down to fbjs?

Probably best to wait and see if any progress is made downstream, otherwise raise awareness of seeking an alternative to packages further down in the chain.

**Comment by johno** [comment]
Thanks for the breakdown @polarthene! Looks like this has now been addressed in #24917 by only installing the specific subpackages we need for GraphQL SDL parsing in Recipes.

The following no longer returns dependency output:

```sh
~/c/gatsby master
❯ yarn why core-js | grep gatsby-recipes
```
